### PR TITLE
fix: migrated add-edit-expense.page to platform [2]

### PR DESCRIPTION
**Pull request by Harshal141**
### Description
Updated the use of the old report model in add edit page to the model form platform API

## Clickup
https://app.clickup.com/t/86cujc9ge

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ import { ExpensePolicy } from 'src/app/core/models/platform/platform-expense-pol
 import { FinalExpensePolicyState } from 'src/app/core/models/platform/platform-final-expense-policy-state.model';
 import { IndividualExpensePolicyState } from 'src/app/core/models/platform/platform-individual-expense-policy-state.model';
 import { PublicPolicyExpense } from 'src/app/core/models/public-policy-expense.model';
-import { UnflattenedReport } from 'src/app/core/models/report-unflattened.model';
+import { PlatformReport } from 'src/app/core/models/platform/platform-report.model';
```
Comment (mvaishnavi) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:73]: naming convention - let's follow folder structure and naming like 
`fyle-mobile-app/src/app/core/models/platform/v1/expense.model.ts`

Comment (Harshal141) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:73]: done

File: src/app/core/mock-data/platform-report.data.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { PlatformReport } from '../models/platform/platform-report.model';
+import { PlatformReport } from '../models/platform/v1/platform-report.model';
 
 export const platformReportData: PlatformReport = {
```
Comment (mvaishnavi) [inline src/app/core/mock-data/platform-report.data.ts:3]: reportData: Report

Comment (mvaishnavi) [reply src/app/core/mock-data/platform-report.data.ts:3]: also rename fyle to `src/app/core/mock-data/platform/v1/report.data.ts`

Comment (mvaishnavi) [reply src/app/core/mock-data/platform-report.data.ts:3]: any reason this data is for a draft zero amount report with no approvals?

Comment (Harshal141) [reply src/app/core/mock-data/platform-report.data.ts:3]: i will update this mock data file when I will create the PR for updating the test for this.

**Review by mvaishnavi — APPROVED** [review]
fix comments, update latest master and resolve any conflicts before reassigning to @Julias0 for review.

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -705,7 +707,7 @@ export class AddEditExpensePage implements OnInit {
           currencyObj: JSON.stringify(this.fg.controls.currencyObj.value),
           fileObjs: JSON.stringify(res.generatedEtxn.dataUrls),
           selectedCCCTransaction: this.selectedCCCTransaction ? JSON.stringify(this.selectedCCCTransaction) : null,
-          selectedReportId: reportValue.report ? JSON.stringify(reportValue.report.rp.id) : null,
+          selectedReportId: reportValue.report ? JSON.stringify(reportValue.report.id) : null,
```
[the remark this answers is not in the document]
Comment (Harshal141) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts]: done

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -3870,15 +3873,15 @@ export class AddEditExpensePage implements OnInit {
...
+              report: Report;
             };
 
             // NOTE: This double call is done as certain fields will not be present in return of upsert call. policy_amount in this case.
             return this.transactionService.upsert(etxn.tx as Transaction).pipe(
               switchMap((txn) => this.transactionService.getETxnUnflattened(txn.id)),
               map((savedEtxn) => savedEtxn && savedEtxn.tx),
               switchMap((tx) => {
-                const selectedReportId = reportControl.report && reportControl.report.rp && reportControl.report.rp.id;
+                const selectedReportId = reportControl.report && reportControl.report && reportControl.report.id;
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3884]: `const selectedReportId = reportControl.report?.id;`

**Review by suyashpatil78 — APPROVED** [review]

**Review by Julias0 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**